apps/typegpu-docs/src/content/docs/fundamentals/slots.mdx

```diff
@@ -67,16 +172,80 @@ export const stepPhysics = tgpu.fn([])(() => {
 ```ts
 // main.ts
 
-import { stepPhysics, getGravitySlot } from './physics';
+import { stepPhysics, gravitySlot } from './physics.ts';
+
+const gravityTowardsCenter = tgpu.fn([vec2f], vec2f)((pos) => {
+  return mul(normalize(pos), -1);
+});
 
-const getGravityTowardsCenter = tgpu
-  .fn([vec2f], vec2f)((position) => mul(normalize(position), -1));
+const stepPhysicsCustomized = stepPhysics
+  .with(gravitySlot, gravityTowardsCenter);
 
-const stepPhysicsAltered = stepPhysics
-  .with(getGravitySlot, getGravityTowardsCenter);
+const main = tgpu['~unstable'].computeFn()(() => {
+  stepPhysicsCustomized(); // <- Will use altered gravity.
+});
+```
 
-const main = tgpu
-  .computeFn()(() => {
-    stepPhysicsAltered(); // <- Will use altered gravity.
-  });
-```
\ No newline at end of file
+{/*
+TODO: re-add and rewrite this section back when we actually prune branches.
+
+## Conditional compilation
+
+Slots can be used for conditional compilation of segments of shaders.
+When the slot holding a boolean value is filled with `false`, all `if` statements depending on the value of that slot will be pruned by our tree-shaking mechanism.
+
+```ts twoslash
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import { max } from 'typegpu/std';
+
+const root = await tgpu.init();
+
+const Obj = d.struct({
+  pos: d.vec3f,
+});
+
+const objects = root.createReadonly(d.arrayOf(Obj, 256));
+
+// ---cut---
+const safeLoopSlot = tgpu.slot(false);
+
+const processObjects = tgpu.fn([])(() => {
+  let len = objects.$.length;
+
+  if (safeLoopSlot.$) { // <- evaluated at compile time
+    // This will be included in the shader code only if
+    // `safeLoopSlot` is bound to true in the caller's scope.
+    len = max(len, 9999);
+  }
+
+  for (let i = 0; i < len; ++i) {
+    // ...
+  }
+});
+```
+
+*/}
+
+## Slots in raw WGSL
+
+It is possible to use slots even in TypeGPU functions that are implemented in WGSL:
+
+```ts twoslash
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+// ---cut---
+const colorSlot = tgpu.slot<string>('vec3f(1., 0., 0.)');
```

Comment:
Let's not use slots for string-code injection like this, even if it's possible 🙈. Use a d.vec3f instead